src/data/store.js

```diff
@@ -0,0 +1,66 @@
+import crystals from './crystals.js';
+import { findByCode } from '../register.js';
+
+const store = {
+    storage: window.localStorage,
+    save(key, item) {
+        const json = JSON.stringify(item);
+        store.storage.setItem(key, json); 
+    },
+    get(key) {
+        const json = store.storage.getItem(key);
+        const item = JSON.parse(json);
+        return item;
+    },
+    getProducts() {
+        let products = store.get('products');
+        if(!products) {
+            store.save('products', crystals);
+            products = crystals;
+        }
+        return products;
+    },
+    getProduct(code) {
+        const products = store.getProducts();
+        const product = findByCode(products, code);
+        return product;
+    },
+    getShoppingCart() {
+        let shoppingCart = store.get('shopping-cart');
+        if(!shoppingCart) {
+            shoppingCart = [];
+        }
+        return shoppingCart;
+    },
+    addToCart(code, quantity) {
+        const shoppingCart = store.getShoppingCart();
+
+        let lineItem = findByCode(shoppingCart, code);
+        if(lineItem) {
+            if(quantity) {
```

Comment:
you don't need this if/else for quantity when the lineItem is already in cart. you can just have:
```
if(lineItem) {
            lineItem.quantity++;
} 
```